fix: fox modal opening from defi and earn opportunity

chxash opened the pull request:
## Description

Fixes fox modal opening in "withdraw" or "deposit" tab instead of "overview".
This behaviour was present from defi page and asset page.
This raises the question of consistency with yearn vaults (modal is still opening in deposit / withdraw page depending on where you are calling it from)

## Notice

<!-- Before submitting a pull request, please make sure you have answered the following: -->

- [ ] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [X] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

https://github.com/shapeshift/web/issues/1656

## Risk

There is a risk of affecting yearn vaults modal opening, I tested it and there is no regression.

## Testing

- Go to fox asset page
- Click on foxy opportunity
- Verify the modal is opened in overview tab

- Go to defi page (+ have some fox staked in foxy)
- Click on foxy opportunity
- Verify the modal is opened in overview tab

## Screenshots (if applicable)

![foxy_defi_modal](https://user-images.githubusercontent.com/5720927/167372623-dfa9efbf-c392-487e-a51c-6ad5ddc4fb40.png)

![defi_yearn](https://user-images.githubusercontent.com/5720927/167372655-e2ca8fc0-9d9f-46b3-88d1-ad5a325b81a4.png)

![earnopp_fox_modal](https://user-images.githubusercontent.com/5720927/167372673-e63cc449-f288-465d-89a7-fae32c70bada.png)

gomesalexandre reviewed (dismissed):
LGTM, tested against develop and this fixes the foxy modal, now opening at overview step.

Could you add a TODO explaining why we're having this condition in the two components? In a world where staking modals are unified (which is the direction we're headed to), we wouldn't need to have such condition and would always open in the overview step (which doesn't yet exist for e.g `vault` staking type).

Also, great concerns about opening on deposit/withdraw depending on where the opportunity row/card is clicked. Tagging @reallybeard here who will be able to provide more insights.

chxash commented:
> LGTM, tested against develop and this fixes the foxy modal, now opening at overview step.
> 
> Could you add a TODO explaining why we're having this condition in the two components? In a world where staking modals are unified (which is the direction we're headed to), we wouldn't need to have such condition and would always open in the overview step (which doesn't yet exist for e.g `vault` staking type).
> 
> Also, great concerns about opening on deposit/withdraw depending on where the opportunity row/card is clicked. Tagging @reallybeard here who will be able to provide more insights.

Added a TODO / comment and fixed the upperCase pathname
